pystiche_papers/ulyanov_et_al_2016/_data.py

```diff
@@ -186,71 +185,74 @@ def images() -> DownloadableImageCollection:
 
 
+class SkipSmallIterableDataset(IterableDataset):
+    def __init__(
+        self,
+        dataset: Dataset,
+        min_size: int,
+        num_samples: int,
+        transform: transforms.Transform,
+    ):
+        self.dataset = dataset
+        self.min_size = min_size
+        self.num_samples = num_samples
+        self.transform = transform
+        self.data_samples = itertools.cycle(self.dataset)
+
+    def __len__(self) -> int:
+        return self.num_samples
+
+    def __iter__(self) -> Generator:
+        num_samples = 0
+        while num_samples < self.num_samples:
+            sample = next(self.data_samples)
+            if all(size >= self.min_size for size in extract_image_size(sample)):
+                # Images that are too small are skipped by the original DataLoader and
+                # the next image is used.
+                # https://github.com/pmeier/texture_nets/blob/aad2cc6f8a998fedc77b64bdcfe1e2884aa0fb3e/dataloader.lua#L91-L100
+                yield self.transform(sample)
+                num_samples += 1
+
+
 def dataset(
     root: str,
     impl_params: bool = True,
     instance_norm: bool = True,
     transform: Optional[transforms.Transform] = None,
-) -> ImageFolderDataset:
+    hyper_parameters: Optional[HyperParameters] = None,
+) -> SkipSmallIterableDataset:
     if transform is None:
         transform = content_transform(
             impl_params=impl_params, instance_norm=instance_norm
         )
-    return ImageFolderDataset(root, transform=transform)
-
-
-def batch_sampler(
-    data_source: Sized,
-    impl_params: bool = True,
-    instance_norm: bool = True,
-    hyper_parameters: Optional[HyperParameters] = None,
-) -> FiniteCycleBatchSampler:
-    r"""Batch sampler from :cite:`ULVL2016,UVL2017`.
-
-    Args:
-        data_source: Dataset to sample from.
-        impl_params: Switch the behavior and hyper-parameters between the reference
-            implementation of the original authors and what is described in the paper.
-            For details see :ref:`here <li_wand_2016-impl_params>`.
-        instance_norm: Switch the behavior and hyper-parameters between both
-            publications of the original authors. For details see
-            :ref:`here <ulyanov_et_al_2016-instance_norm>`.
-        hyper_parameters: Hyper parameters. If omitted,
-            :func:`~pystiche_papers.ulyanov_et_al_2016.hyper_parameters` is used.
-    """
     if hyper_parameters is None:
         hyper_parameters = _hyper_parameters(
             impl_params=impl_params, instance_norm=instance_norm
         )
-
-    return FiniteCycleBatchSampler(
-        data_source,
-        num_batches=hyper_parameters.batch_sampler.num_batches,
-        batch_size=hyper_parameters.batch_sampler.batch_size,
+    min_size = hyper_parameters.content_transform.edge_size
+    num_samples = (
+        hyper_parameters.sampler.num_samples * hyper_parameters.sampler.batch_size
+    )
```

Comment:
Something is off with the naming here. `num_samples` shouldn't be `num_samples` times anything. Maybe rename the one in the hyperparameters back to `num_batches`?